### DISCUSSION TITLE
Still have many small files in the arctic table after optimizing

### DISCUSSION THREAD

**Discussion by xxubai** [apache/amoro#605] [Q&A] [answered]
Hi team, I run flink job and sink data into arctic(do a checkpoint every 30 minutes), but I find that still have many small files even though have optimized. How could I reduce small files? For example: one partition has only one base files(with some pos-delete files) that size is smaller than 256M.
I changed the table properties,  but it doesn't seem to be working as I expected.

|optimize.major.trigger.max-interval | 3600000|
|------|------|
|optimize.minor.trigger.max-interval | 180000|

**Comment by xxubai** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/54210725/201870889-19cc7eef-e050-40e4-b53b-ab976e4b095e.png)
![image](https://user-images.githubusercontent.com/54210725/201871095-65cea9f6-188a-4ee4-9a93-498fcd2ccbb0.png)

**Reply by wangtaohz** [reply]
> ![image](https://user-images.githubusercontent.com/54210725/201870889-19cc7eef-e050-40e4-b53b-ab976e4b095e.png) ![image](https://user-images.githubusercontent.com/54210725/201871095-65cea9f6-188a-4ee4-9a93-498fcd2ccbb0.png)

Are these small files in base location or change location？

**Reply by xxubai** [reply]
They both contain small files
![image](https://user-images.githubusercontent.com/54210725/202067503-449140e6-a95d-47ad-a31e-ea8ff8565fa3.png)

**Reply by wangtaohz** [reply]
It's reasonable that there are only 28 files (of current snapshot) in base store. If you still see many files in your file system, these files most likely belong to old snapshots which have are not expired.

Arctic base store keeps snapshots in latest 12 hours by default, to keep fewer snapshots, you can change the table property `snapshot.base.keep.minutes`(more info https://arctic.netease.com/ch/meta-service/table-properties/#_5)

**Reply by wangtaohz** [reply]
In change store, similar to base store，you can change these properties, `change.data.ttl.minutes`, `snapshot.change.keep.minutes`.

**Reply by xxubai** [reply]
That works! Thanks.

**Reply by xxubai** [reply]
![image](https://user-images.githubusercontent.com/54210725/202112398-70239885-2cdb-4f39-88c6-6763ae6e2894.png)
![image](https://user-images.githubusercontent.com/54210725/202112477-e6b46285-335b-452f-879c-1e868c586663.png)

**Comment by wangtaohz** [comment]
And what is the table type？
If it is upgraded from a hive table, maybe you should set `optimize.full.trigger.max-interval`, to trigger the full optimize, which will rewrite all base files.

**Reply by xxubai** [reply]
The table type is arctic
